### PR TITLE
Update go mod hcsshim version to fix the kube-proxy 1.15 issue 

### PR DESCRIPTION
**Pull request by wawa0210**
Update go mod hcsshim version to fix the kube-proxy issue cannot access service by self nodeip:port  on windows

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:

/kind bug

**What this PR does / why we need it**:
fix the go mod import error, that's import a old module will cause some problems in windows kube-proxy. Such as cannot access service by self nodeip:port on kube-proxy 1.15.0

**Which issue(s) this PR fixes**:
https://github.com/kubernetes/kubernetes/issues/79515

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
update hcsshim repo version to fix kube-proxy bug on windows
```
/assign @feiskyer

### CODE REVIEW

**Comment by dims** [comment]
/hold

@wawa0210 can you please ask MSFT folks to cut a new release v0.8.7 with the changes you need? (we want to avoid using a random SHA)

**Comment by wawa0210** [comment]
> /hold
> 
> @wawa0210 can you please ask MSFT folks to cut a new release v0.8.7 with the changes you need? (we want to avoid using a random SHA)

I will try that, thx

**Comment by justaugustus** [comment]
@craiglpeters @lachie83 @jhendrixMSFT

File: go.mod
Comment (andyzhangx) [inline go.mod]: the hcsshim version `v0.0.0-20190417211021-672e52e9209d` looks strange, the fixed version should be bigger than `v0.8.6`, right?

Comment (andyzhangx) [reply go.mod]: also you need to run following command to update vendor files, only go mod change won't fix this issue.
```
export GO111MODULE=on
go mod vendor
```

Comment (BenTheElder) [reply go.mod]: `hack/update-vendor.sh` will handle the `go mod vendor` and other necessary fixups.

Comment (wawa0210) [reply go.mod]: Yeah,It's should be bigger than v0.8.6 but the hcsshim repo has not a new tag version

Comment (wawa0210) [reply go.mod]: > the hcsshim version `v0.0.0-20190417211021-672e52e9209d` looks strange, the fixed version should be bigger than `v0.8.6`, right?

Yeah,It's should be bigger than v0.8.6 but the hcsshim repo has not a new tag version

I don't know why the  pull-kubernetes-dependencies was failed , can you help me ？ any suggestion about that ?

Comment (wawa0210) [reply go.mod]: > the hcsshim version `v0.0.0-20190417211021-672e52e9209d` looks strange, the fixed version should be bigger than `v0.8.6`, right?

Yeah,It's should be bigger than v0.8.6 but the hcsshim repo has not a new tag version

You mean is that I should run go mod vendor and commit the hcsshim changed files ?

Comment (andyzhangx) [reply go.mod]: @wawa0210 I think we should wait for a new release of `github.com/Microsoft/hcsshim`
about the bazel error, you need to run `hack/update-bazel.sh`

Comment (BenTheElder) [reply go.mod]: No worries.
So if you click [the job failure link](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/80116/pull-kubernetes-dependencies/1149994565093036034) from below and expand either the full logs or the "vendor" failure in the junit you will see the diff that caused failure.

At the bottom of that it tells you to run `hack/update-vendor.sh` to fix the failure.

@andyzgangx there's some other non-trivial logic in update-vendor besides go mod vendor and update-bazel (dependency set pruning in particular). It's best to run that tool 😅

Comment (wawa0210) [reply go.mod]: @andyzhangx  Because the previous version of this repository (v0.8.6) was still five months ago, the code after v0.8.6 was referenced in kubernetes1.14.3.

Comment (wawa0210) [reply go.mod]: @BenTheElder  i run hack/update-vendor.sh then many other files had been changed ,should i pull all the files have been changed?

Comment (feiskyer) [reply go.mod]: Should probable also run `hack/pin-dependency.sh $MODULE $SHA-OR-TAG` to pin hcsshim to the new version

Comment (wawa0210) [reply go.mod]: @feiskyer  Thanks a lot , when i run the script `hack/pin-dependency.sh $MODULE $SHA-OR-TAG` and push again , all the checks passed

**Review by BenTheElder — APPROVED** [review]
vendor looks sane to me
/approve

File: go.mod
Comment (feiskyer) [reply go.mod]: 👍

**Review by feiskyer — COMMENTED** [review]
/lgtm

**Comment by wawa0210** [comment]
> /hold
> 
> @wawa0210 can you please ask MSFT folks to cut a new release v0.8.7 with the changes you need? (we want to avoid using a random SHA)

Hey, i already asked the MSFT folks request for a new version , but didn't get any feedback。

https://github.com/microsoft/hcsshim/issues/642

**Comment by feiskyer** [comment]
Since there're no new releases yet in hcsshim, I think it's ok to use the commit id in the vendors. @wojtek-t could you help to approve the changes?

**Comment by PatrickLang** [comment]
Other projects (containerd/containerd) also use a commit it

https://github.com/containerd/containerd/blob/86442dfbb9c76d20031718d471c0d5b33c938ef3/vendor.conf#L37

**Comment by dims** [comment]
@PatrickLang not disputing that. 

question(s) are as follows, do we know anyone in that project? Do they think the newer SHA(s) we use are good for us to use? If so, is there a reason they have not tagged it yet. could they tag it so we know they are ok with us using it? If we don't have anything who is responsive in that repo, should we even continue to use it? is there an alternative?

This is part of the larger effort to move us off random SHA(s) https://github.com/kubernetes/kubernetes/issues/79234

**Comment by PatrickLang** [comment]
/assign @PatrickLang 

I'll get a test pass run on this then remove the hold

**Comment by PatrickLang** [comment]
There aren't any tags on hcsshim because it's not tested or released as an isolated project. Changes are accumulated there to fix containerd and moby bugs, then vendored into those projects. I think we should be able to remove this direct dependency later once we can move onto CRI-ContainerD and deprecate dockershim.

**Comment by wawa0210** [comment]
/assign @wojtek-t

Can you review this pr and approve that?

**Comment by liggitt** [comment]
> There aren't any tags on hcsshim because it's not tested or released as an isolated project

Did something change in the project? releases used to be tagged regularly (see https://github.com/microsoft/hcsshim/tags)

If no more tags/releases are planned, it might be worth clarifying that in the project readme. Having tags/releases is a strong signal that those tags should be used as stable targets.

File: go.mod
Comment (liggitt) [reply go.mod]: > @andyzhangx Because the previous version of this repository (v0.8.6) was still five months ago, the code after v0.8.6 was referenced in kubernetes1.14.3.

I'm not sure what you're referring to. 1.14 is using 69ac8d3f7fc10a0623f3a2655958a1a5bb71f58f, which is pre-v0.8.4

See https://github.com/kubernetes/kubernetes/blob/release-1.14/Godeps/Godeps.json#L157-L161

```sh
$ pwd
/Users/liggitt/go/src/github.com/Microsoft/hcsshim

$ git tag --contains 69ac8
v0.8.4
v0.8.5
v0.8.6
```

Comment (wawa0210) [reply go.mod]: > > @andyzhangx Because the previous version of this repository (v0.8.6) was still five months ago, the code after v0.8.6 was referenced in kubernetes1.14.3.
> 
> I'm not sure what you're referring to. 1.14 is using 69ac8d3f7fc10a0623f3a2655958a1a5bb71f58f, which is pre-v0.8.4
> 
> See https://github.com/kubernetes/kubernetes/blob/release-1.14/Godeps/Godeps.json#L157-L161
> 
> ```shell
> $ pwd
> /Users/liggitt/go/src/github.com/Microsoft/hcsshim
> 
> $ git tag --contains 69ac8
> v0.8.4
> v0.8.5
> v0.8.6
> ```

A commit version after v0.8.6 was used before the introduction of go mod. But later, due to the introduction of go mod, the hcsshim version has returned to v0.8.6, which will cause kube-proxy to not work on a specific version of windows server 2019.

**Comment by wawa0210** [comment]
> > There aren't any tags on hcsshim because it's not tested or released as an isolated project
> 
> Did something change in the project? releases used to be tagged regularly (see https://github.com/microsoft/hcsshim/tags)
> 
> If no more tags/releases are planned, and downstream projects should always use `master`, it might be worth clarifying that in the project readme. Having tags/releases is a strong signal that those tags should be used as stable targets.

A commit after v0.8.6 was used before the introduction of go mod

> > There aren't any tags on hcsshim because it's not tested or released as an isolated project
> 
> Did something change in the project? releases used to be tagged regularly (see https://github.com/microsoft/hcsshim/tags)
> 
> If no more tags/releases are planned, and downstream projects should always use `master`, it might be worth clarifying that in the project readme. Having tags/releases is a strong signal that those tags should be used as stable targets.

**Comment by wawa0210** [comment]
r

File: go.mod
Comment (liggitt) [reply go.mod]: > A commit version after v0.8.6 was used before the introduction of go mod.

Can you give a link to the kubernetes branch and hcsshim commit you're referring to? The only changes I see to that dependency are at https://github.com/kubernetes/kubernetes/commits/master/vendor/github.com/Microsoft/hcsshim

* a4272099e72ebdaa03da52b98fd28ea861ba0bf9 updated to 69ac8d3f7fc10a0623f3a2655958a1a5bb71f58f, which is pre-v0.8.4
* d0261b10770210f83edbcfe379db24e1b82a9e86 switched to go modules and stayed at the same revision
* c32c26000d68fc0f68d4da920c82ad8701808066 switched to v0.8.6

Comment (liggitt) [reply go.mod]: To be clear, I'm not opposed to updating to the current sha if the intent of the hcsshim project is to be used that way... just making sure we're all on the same page that this appears to be a shift in how we're consuming the project, not fixing a regression.

**Comment by wawa0210** [comment]
I guess the change is that : https://github.com/microsoft/hcsshim/commit/672e52e9209d1e53718c1b6a7d68cc9272654ab5

The new commit has fix this

在 2019年9月24日，上午12:25，Jordan Liggitt <notifications@github.com<mailto:notifications@github.com>> 写道：


@liggitt commented on this pull request.

________________________________

In go.mod<https://github.com/kubernetes/kubernetes/pull/80116#discussion_r327209035>:

> @@ -14,7 +14,7 @@ require (
        github.com/JeffAshton/win_pdh<http://github.com/JeffAshton/win_pdh> v0.0.0-20161109143554-76bb4ee9f0ab
        github.com/MakeNowJust/heredoc<http://github.com/MakeNowJust/heredoc> v0.0.0-20170808103936-bb23615498cd
        github.com/Microsoft/go-winio<http://github.com/Microsoft/go-winio> v0.4.11
-       github.com/Microsoft/hcsshim<http://github.com/Microsoft/hcsshim> v0.8.6
+       github.com/Microsoft/hcsshim<http://github.com/Microsoft/hcsshim> v0.0.0-20190417211021-672e52e9209d


A commit version after v0.8.6 was used before the introduction of go mod.

Can you give a link to the kubernetes branch and hcsshim commit you're referring to? The only changes I see to that dependency are at https://github.com/kubernetes/kubernetes/commits/master/vendor/github.com/Microsoft/hcsshim

  *   a427209<https://github.com/kubernetes/kubernetes/commit/a4272099e72ebdaa03da52b98fd28ea861ba0bf9> updated to 69ac8d3f7fc10a0623f3a2655958a1a5bb71f58f, which is pre-v0.8.4
  *   d0261b1<https://github.com/kubernetes/kubernetes/commit/d0261b10770210f83edbcfe379db24e1b82a9e86> switched to go modules and stayed at the same revision
  *   c32c260<https://github.com/kubernetes/kubernetes/commit/c32c26000d68fc0f68d4da920c82ad8701808066> switched to v0.8.6

**Comment by wawa0210** [comment]
> > There aren't any tags on hcsshim because it's not tested or released as an isolated project
> 
> Did something change in the project? releases used to be tagged regularly (see https://github.com/microsoft/hcsshim/tags)
> 
> If no more tags/releases are planned, and downstream projects should always use `master`, it might be worth clarifying that in the project readme. Having tags/releases is a strong signal that those tags should be used as stable targets.



> > There aren't any tags on hcsshim because it's not tested or released as an isolated project
> 
> Did something change in the project? releases used to be tagged regularly (see https://github.com/microsoft/hcsshim/tags)
> 
> If no more tags/releases are planned, and downstream projects should always use `master`, it might be worth clarifying that in the project readme. Having tags/releases is a strong signal that those tags should be used as stable targets.

image
More stringent judgment conditions have been added after v0.8.6, so that this situation no longer occurs

File: go.mod
Comment (wawa0210) [reply go.mod]: > To be clear, I'm not opposed to updating to the current sha if the intent of the hcsshim project is to be used that way... just making sure we're all on the same page that this appears to be a shift in how we're consuming the project, not fixing a regression.

More stringent judgment conditions have been added after v0.8.6, so that this situation no longer occurs

https://github.com/microsoft/hcsshim/commit/672e52e9209d1e53718c1b6a7d68cc9272654ab5

image

I guess this problem may occur before v0.8.6.

**Comment by liggitt** [comment]
hmm, looks like this was accidentally closed? mind reopening it?

**Comment by wawa0210** [comment]
> hmm, looks like this was accidentally closed? mind reopening it?

I am very sorry, because this pull request has been around for a long time, I used to mistakenly delete the fork repo. Just after turning off pr by mistake, I don’t know how to reopen it now. ops

**Comment by liggitt** [comment]
np, I picked it and restored it at https://github.com/kubernetes/kubernetes/pull/83027

**Comment by wawa0210** [comment]
> no problem, I picked it and restored it at #83027

Can I re-raise a pr, I hope to be a contributor

**Comment by liggitt** [comment]
> Can I re-raise a pr, I hope to be a contributor

the commit is still attributed to you (see https://patch-diff.githubusercontent.com/raw/kubernetes/kubernetes/pull/83027.patch)

**Comment by wawa0210** [comment]
> > Can I re-raise a pr, I hope to be a contributor
> 
> the commit is still attributed to you (see https://patch-diff.githubusercontent.com/raw/kubernetes/kubernetes/pull/83027.patch)

By contributing code, I can get tickets for Shanghai kubecon next year. 😄😄😄

**Comment by wawa0210** [comment]
> > Can I re-raise a pr, I hope to be a contributor
> 
> the commit is still attributed to you (see https://patch-diff.githubusercontent.com/raw/kubernetes/kubernetes/pull/83027.patch), though if you want to pick that and open a PR yourself, that's fine too

Does this pr require hot fixes to other distributions such as 1.5.x and 1.6.x  ？

**Comment by liggitt** [comment]
> Does this pr require hot fixes to other distributions such as 1.5.x and 1.6.x ？

Only the 1.14.x+ branches are currently maintained. I'll defer to sig-windows / sig-network whether the issue is severe enough and the fix non-invasive enough to justify cherry-picking to those branches after merge

**Comment by wawa0210** [comment]
> > Does this pr require hot fixes to other distributions such as 1.5.x and 1.6.x ？
> 
> Only the 1.14.x+ branches are currently maintained. I'll defer to sig-windows / sig-network whether the issue is severe enough and the fix non-invasive enough to justify cherry-picking to those branches after merge

Just typing incorrectly, I am referring to the 1.15.x and 1.16.x versions. These two versions have my own compiled and tested in the local environment, kube-proxy can work normally.

**Comment by liggitt** [comment]
this commit was reopened/merged in https://github.com/kubernetes/kubernetes/pull/83027

**Pull request closed without merging**
